### PR TITLE
Storage/File Share test improvement

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by microzchang** [comment]
It's really difficult to test file handles...

File: sdk/storage/azure-storage-files-shares/test/ut/share_sas_test.cpp [hunk 1]
```diff
@@ -9,107 +9,228 @@
...
+  public:
+    template <class T>
+    T GetSasAuthenticatedClient(const T& shareClient, const std::string& sasToken)
+    {
+      T fileClient1(
+          AppendQueryParameters(Azure::Core::Url(shareClient.GetUrl()), sasToken),
+          InitStorageClientOptions<Files::Shares::ShareClientOptions>());
+      return fileClient1;
+    }
+    void VerifyDataLakeSasRead(
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/test/ut/share_sas_test.cpp:22]: This is not datalake, it's file share

File: sdk/storage/azure-storage-files-shares/test/ut/share_sas_test.cpp [hunk 2]
```diff
@@ -9,107 +9,228 @@
...
+
+    void VerifyDataLakeSasList(
+        const Files::Shares::ShareDirectoryClient& directoryClient,
+        const std::string& sasToken)
+    {
+      auto directoryClient1 = GetSasAuthenticatedClient(directoryClient, sasToken);
+      EXPECT_NO_THROW(directoryClient1.ListFilesAndDirectories());
+    }
+
+    void VerifyDataLakeSasMove(
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/test/ut/share_sas_test.cpp:72]: Does share service support moving file?

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**